pkg/compiler/func_scope.go

```diff
@@ -85,7 +85,19 @@ func (c *codegen) getFuncNameFromDecl(pkgPath string, decl *ast.FuncDecl) string
 		case *ast.Ident:
 			name = t.Name + "." + name
 		case *ast.StarExpr:
-			name = t.X.(*ast.Ident).Name + "." + name
+			switch t.X.(type) {
+			case *ast.Ident:
+				name = t.X.(*ast.Ident).Name + "." + name
+			case *ast.IndexExpr:
+				// Generic func declaration receiver: func (x *Pointer[T]) Load() *T
+				name = t.X.(*ast.IndexExpr).X.(*ast.Ident).Name + "." + name
+			}
+		case *ast.IndexExpr:
+			switch t.X.(type) {
```

Comment:
Why a switch here?

Reply:
Yeah, there needs to be some reasonable error, otherwise we have an undefined behavior and that's not what I would like to have in the compiler.